Makefile

```diff
@@ -12,27 +14,40 @@
 #See the License for the specific language governing permissions and
 #limitations under the License.
 
-CXXFLAGS = -c -std=c++17 -O2 -Wall -Wextra -Werror -g
+ifeq (${DEBUG},1)
+	CXXFLAGS = -std=c++17 -O0 -Wall -Wextra -Werror -g
+else
+	CXXFLAGS = -std=c++17 -O2 -Wall -Wextra -Werror
+endif
+
 LDLIBS = -lodbc -lpthread
 
+SRCDIR = src
+BUILDDIR = build
+
 SOURCES = \
-	src/AnalyticalStatistic.cc \
-	src/chBenchmark.cc \
-	src/DataSource.cc \
-	src/DbcTools.cc \
-	src/dialect/DialectStrategy.cc \
-	src/Log.cc \
-	src/PthreadShim.cc \
-	src/Queries.cc \
-	src/Random.cc \
-	src/Schema.cc \
-	src/TransactionalStatistic.cc \
-	src/Transactions.cc \
-	src/TupleGen.cc
-
-chBenchmark: $(SOURCES:.cc=.o)
-	$(CXX) $(LDFLAGS) $^ $(LDLIBS) -o $@
+	$(SRCDIR)/AnalyticalStatistic.cc \
+	$(SRCDIR)/chBenchmark.cc \
+	$(SRCDIR)/DataSource.cc \
+	$(SRCDIR)/DbcTools.cc \
+	$(SRCDIR)/dialect/DialectStrategy.cc \
+	$(SRCDIR)/Log.cc \
+	$(SRCDIR)/PthreadShim.cc \
+	$(SRCDIR)/Queries.cc \
+	$(SRCDIR)/Random.cc \
+	$(SRCDIR)/Schema.cc \
+	$(SRCDIR)/TransactionalStatistic.cc \
+	$(SRCDIR)/Transactions.cc \
+	$(SRCDIR)/TupleGen.cc
+
+$(BUILDDIR)/%.o : $(SRCDIR)/%.cc
+	mkdir -p $(@D) && \
+	$(CXX) $(CPPFLAGS) $(CXXFLAGS) -c $^ -o $@
+
+OBJS = $(subst $(SRCDIR),$(BUILDDIR),$(SOURCES:.cc=.o))
+
+$(BUILDDIR)/chBenchmark: $(OBJS)
+	$(CXX) $(LDFLAGS) $(OBJS) $(LDLIBS) -o $@
```

Comment:
If you set the default linker for `.o` files to `$(LINK.cc)`, then you can avoid specifying a rule body here at all:

```make
LINK.o = $(LINK.cc)

$(BUILDDIR)/chBenchmark: $(OBJS)
```